opentracing-cdi/src/main/java/io/opentracing/contrib/cdi/internal/TracerInitializer.java

```diff
@@ -0,0 +1,60 @@
+package io.opentracing.contrib.cdi.internal;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.tracerresolver.TracerResolver;
+import io.opentracing.util.GlobalTracer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import java.util.logging.Logger;
+
+/**
+ * This is not part of the "business" of this example, but can be used as an example of how to register a Tracer
+ * with the GlobalTracer. On some platforms, like Wildfly Swarm, this is done automatically for you.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+public class TracerInitializer {
+    private static final Logger log = Logger.getLogger(TracerInitializer.class.getName());
+
+    @Inject
+    Instance<Tracer> tracerInstance;
+
+    public void init(@Observes @Initialized(ApplicationScoped.class) Object ignored) {
+        // 1) check if there's a Tracer that can be injected
+        // 1.1) if we have one, and if it's *not* our own producer returning a GlobalTracer, then it's a user's configured
+        //      and we'll just use it
+        // 2) check if GlobalTracer already has a Tracer registered
+        // 3) use the TracerResolver to get a Tracer
+        // 4) register this with the GlobalTracer
+
+        if (!tracerInstance.isUnsatisfied()) {
+            // the Tracer is already available!
+            Tracer tracer = tracerInstance.get();
+
+            if (!(tracer instanceof GlobalTracer)) {
+                // it's not our own producer, which returns a GlobalTracer. We'll just use it then!
+                GlobalTracer.register(tracer);
```

Comment:
May be better to move this to a separate artifact under the java-cdi repo, related to tracer resolution. So then there would be artifact for instrumentation of CDI components, and a separate artifact to provide access to a tracer via CDI.